pkg/vmprovider/providers/vsphere2/virtualmachine/configspec_test.go

```diff
@@ -142,6 +146,114 @@ var _ = Describe("CreateConfigSpec", func() {
 				Expect(configSpec.Firmware).ToNot(Equal(vm.Annotations[constants.FirmwareOverrideAnnotation]))
 			})
 		})
+
+		When("VM Image hardware version is set", func() {
+			BeforeEach(func() {
+				vmImageStatus.HardwareVersion = pointer.Int32(101)
+			})
+
+			It("config spec has expected version", func() {
+				Expect(configSpec.Version).To(Equal("vmx-101"))
```

Comment:
Nit: Add comment for Dummy VM having a PVC --  the image Version gets considered because of it.